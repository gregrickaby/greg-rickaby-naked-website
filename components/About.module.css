.about {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 740px;
  padding: 0 1.2rem;
}

@media (min-width: 768px) {
  .about {
    padding: 0;
  }
}
