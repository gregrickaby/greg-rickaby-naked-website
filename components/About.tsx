import styles from "./About.module.css";

export default function About() {
  return (
    <div id="about" className={styles.about}>
      <h2>About</h2>
      <p>
        Way back in 1997, I built my first website on Geocities and have been
        passionate about the web since! In my spare time I enjoy{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          taking photos
        </a>
        , cheering for the{" "}
        <a href="https://www.packers.com/">Green Bay Packers</a>, and making
        pizzas.
      </p>

      <h3>Web Developer</h3>
      <p>
        I&apos;m a Tech Lead at{" "}
        <a href="https://americaneagle.com/">AmericanEagle.com</a>, an industry
        leader in web design, development, hosting, and digital marketing. I
        partner with clients and team members to build websites with Next.js and
        WordPress.
      </p>
      <p>
        View my r√©sum√© on{" "}
        <a href="https://www.linkedin.com/in/gregrickaby/" rel="author">
          LinkedIn
        </a>
        .
      </p>

      <h3>Published Author</h3>
      <p>
        In 2017, I wrote a children&apos;s book titled{" "}
        <a href="https://amzn.to/34QAxAQ">
          <em>
            <strong>Creating a Website for Dummies Jr</strong>
          </em>
        </a>{" "}
        which was published by{" "}
        <a href="https://www.wiley.com/en-us"> Wiley Global</a> under the{" "}
        <a href="https://www.dummies.com/">dummies‚Ñ¢Ô∏è</a> brand.
      </p>

      <p>
        The book guides young readers through the planning and development
        processes of building a website. It has been translated into other 2
        other languages, and currently has a 4.5-star rating on Amazon.
      </p>

      <p>
        Since being published, I&apos;ve also served as the Technical Editor on
        three other books:
      </p>
      <ul>
        <li>
          <a href="https://amzn.to/37XCflU" rel="nofollow">
            <em>WordPress for Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2021)
        </li>
        <li>
          <a href="https://amzn.to/2BgCg7i" rel="nofollow">
            <em>Professional WordPress Plugin Development</em>
          </a>{" "}
          (Williams et al., 2020)
        </li>
        <li>
          <a href="https://amzn.to/37MMDLp" rel="nofollow">
            <em>WordPress All-In-One For Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2019)
        </li>
      </ul>

      <h3>Photographer</h3>
      <p>
        In late 2019, I bought a mirrorless camera and spent months taking
        courses, reading, practicing, and having my work critiqued by peers.
        What has started as an enjoyable hobby, has turned into a passionate way
        to express my creativity.
      </p>
      <p>
        View my{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          portfolio
        </a>{" "}
        on SmugMug.
      </p>

      <h3>Contributions</h3>
      <p>
        I've been part of the WordPress community since 2008. I've made
        contributions to core, documentation, plugins, and themes. I've also
        spoken at WordCamps and meet-ups. I'm also a contributor to other
        open-source projects like Next.js, Storybook, and Gatsby.
      </p>
      <p>
        Outside of tech, I volunteer my time and efforts at local community
        organizations. The Boy Scouts, Wiregrass Church, and the Enterprise High
        School Band Boosters to name a few.
      </p>

      <p>
        You can find me around the web with the links below. Stay safe out
        there, and thanks for reading!
        <span role="img" aria-label="cheers">
          üçª
        </span>
      </p>
    </div>
  );
}
