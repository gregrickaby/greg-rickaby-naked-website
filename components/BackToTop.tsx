'use client'

import {IconArrowUp} from '@tabler/icons-react'
import {useEffect, useState} from 'react'

/**
 * The back to top component.
 */
export default function BackToTop() {
  const [showButton, setShowButton] = useState(false)
  const buttonText = 'Go back to the top of the page'

  /**
   * Scroll handler.
   */
  const handleScroll = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    })
  }

  /**
   * Effect for showing the back to top button.
   */
  useEffect(() => {
    // Scroll event handler.
    const handleScroll = () => {
      setShowButton(window.scrollY > 200)
    }

    // Add event listener.
    window.addEventListener('scroll', handleScroll)

    // Cleanup event listener.
    return () => window.removeEventListener('scroll', handleScroll)
  }, [])

  return showButton ? (
    <button
      aria-label={buttonText}
      className="button fixed bottom-8 right-3 rounded-full bg-gray-400 p-2 text-white shadow-lg hover:bg-gray-500 lg:right-12 dark:bg-gray-800 hover:dark:bg-gray-700"
      data-testid="back-to-top"
      onClick={handleScroll}
      title={buttonText}
    >
      <IconArrowUp height="32" width="32" />
      <span className="sr-only">{buttonText}</span>
    </button>
  ) : null
}
