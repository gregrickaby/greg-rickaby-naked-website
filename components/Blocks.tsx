'use client'

import '@fancyapps/ui/dist/fancybox/fancybox.css'
import {useEffect, useRef} from 'react'

/**
 * The blocks component.
 */
export default function Blocks({content}: Readonly<{content: string}>) {
  // Create a ref for the post content.
  const contentRef = useRef<HTMLDivElement>(null)

  /**
   * Initialize photo gallery.
   */
  async function galleryInit() {
    // On the server? Bail.
    if (typeof window === 'undefined') {
      return
    }

    // No ref? Bail.
    if (!contentRef.current) {
      return
    }

    // Attempt to find the gallery.
    const gallery = contentRef.current.querySelector('.grd-photo-gallery-grid')

    // No gallery? Bail.
    if (!gallery) {
      return
    }

    // Import the libraries on demand.
    const Masonry = (await import('masonry-layout')).default
    const Fancybox = (await import('@fancyapps/ui')).Fancybox
    const imagesLoaded = (await import('imagesloaded')).default

    // Initialize ImagesLoaded.
    const imagesLoadedInstance = imagesLoaded(gallery)

    // Initialize Masonry.
    const masonryGrid = new Masonry(gallery, {
      gutter: 24,
      itemSelector: '.grd-photo-gallery-image',
      columnWidth: '.grd-photo-gallery-grid-sizer',
      percentPosition: true
    })

    // When ImagesLoaded is done, initialize Masonry.
    imagesLoadedInstance.on('done', () => {
      masonryGrid?.layout?.()
    })

    // Initialize Fancybox.
    Fancybox.bind('[data-fancybox]', {
      animated: false,
      groupAll: true,
      defaultDisplay: 'flex',
      Toolbar: {
        display: {
          left: ['infobar'],
          middle: [
            'zoomIn',
            'zoomOut',
            'toggle1to1',
            'rotateCCW',
            'rotateCW',
            'flipX',
            'flipY'
          ],
          right: ['slideshow', 'thumbs', 'download', 'close']
        }
      }
    })

    // Destroy Masonry and Fancybox when the component unmounts.
    return () => {
      masonryGrid?.destroy?.()
      Fancybox.destroy?.()
    }
  }

  /**
   * Initialize the gallery when the component mounts.
   */
  useEffect(() => {
    galleryInit()
  }, [content])

  return (
    <div
      ref={contentRef}
      className="text-lg"
      dangerouslySetInnerHTML={{__html: content}}
    />
  )
}
