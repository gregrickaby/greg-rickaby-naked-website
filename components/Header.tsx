import config from '@/lib/config'
import Image from 'next/image'
import Link from 'next/link'
import Logo from '../app/icon.png'

/**
 * Header component.
 */
export default function Header() {
  return (
    <header
      className="flex flex-col items-center justify-center gap-12 text-center md:flex-row md:justify-between"
      data-testid="header"
    >
      <div className="not-prose flex flex-col items-center gap-4 md:flex-row md:text-left">
        <Link href="/">
          <Image
            alt="Greg Rickaby"
            className="mx-auto h-24 w-24 rounded-full shadow-lg"
            height={96}
            loading="eager"
            src={Logo}
            width={96}
          />
        </Link>
        <div className="flex flex-col md:text-left">
          <Link href="/" className="no-underline">
            <h1 className="mb-3 text-4xl font-bold leading-none">
              {config.siteName}
            </h1>
          </Link>
          <p className="m-0 max-w-sm p-0 text-lg">{config.siteDescription}</p>
        </div>
      </div>
    </header>
  )
}
