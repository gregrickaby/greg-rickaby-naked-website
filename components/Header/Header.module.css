.header {
  @apply mb-16 flex items-center justify-between gap-12 p-8 text-center;
  @apply md:flex-row md:justify-between md:p-12;
}

.headerContainer {
  @apply flex flex-row items-center gap-4 lg:flex-row lg:text-left;
}

.logo {
  @apply mx-auto rounded-full shadow-lg transition-transform duration-300 ease-in-out;
  @apply hover:rotate-45 motion-reduce:hover:transform-none;
}

.brandingContainer {
  @apply flex flex-col text-left;
}

.siteTitle {
  @apply mb-1 text-2xl font-bold leading-none tracking-tighter md:text-4xl;
}

.siteDescription {
  @apply m-0 max-w-sm p-0 md:ml-1 md:text-lg;
}
