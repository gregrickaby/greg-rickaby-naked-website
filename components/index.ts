export * from './ArticleCard'
export * from './BackToTop'
export * from './Blocks'
export * from './BlogArchive'
export * from './Byline'
export * from './CommentForm'
export * from './Comments'
export * from './Footer'
export * from './Header'
export * from './JsonLd'
export * from './Navigation'
export * from './SocialLinks'
