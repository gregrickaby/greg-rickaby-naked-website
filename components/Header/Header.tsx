import Logo from '@/app/icon.png'
import {Navigation} from '@/components/Navigation'
import config from '@/lib/config'
import Image from 'next/image'
import Link from 'next/link'
import styles from './Header.module.css'

/**
 * Header component.
 */
export function Header() {
  return (
    <header className={styles.header} data-testid="header">
      <div className={styles.headerContainer}>
        <Link href="/">
          <Image
            alt="Greg Rickaby"
            className={styles.logo}
            height={96}
            loading="eager"
            src={Logo}
            width={96}
          />
        </Link>
        <div className={styles.brandingContainer}>
          <h1 className={styles.siteTitle}>
            <Link href="/">{config.siteName}</Link>
          </h1>
          <p className={styles.siteDescription}>{config.siteDescription}</p>
        </div>
      </div>
      <Navigation />
    </header>
  )
}
