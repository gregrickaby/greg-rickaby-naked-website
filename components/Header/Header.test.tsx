import {Header} from '@/components/Header'
import config from '@/lib/config'
import {act, render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {axe} from 'jest-axe'

/**
 * Test suite for the Header component.
 */
describe('Header', () => {
  it('should render', () => {
    render(<Header />)
    const header = screen.getByTestId('header')
    expect(header).toBeInTheDocument()
  })

  it('should render the logo', () => {
    render(<Header />)
    const logo = screen.getByAltText(config.siteName)
    expect(logo).toBeInTheDocument
  })

  it('should render the site name', () => {
    render(<Header />)
    const text = screen.getByText(config.siteName)
    expect(text).toBeInTheDocument()
  })

  it('should render the description text', () => {
    render(<Header />)
    const text = screen.getByText(config.siteDescription)
    expect(text).toBeInTheDocument
  })

  it('should not have accessibility issues', async () => {
    const {container} = render(<Header />)
    const button = screen.getByRole('button')

    await userEvent.click(button)

    const results = await act(async () => {
      return await axe(container)
    })

    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    render(<Header />)
    const header = screen.getByTestId('header')
    expect(header).toMatchSnapshot()
  })
})
