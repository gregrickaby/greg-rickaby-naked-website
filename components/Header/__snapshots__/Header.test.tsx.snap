// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match snapshot 1`] = `
<header
  class="_header_46c4bf"
  data-testid="header"
>
  <div
    class="_headerContainer_46c4bf"
  >
    <a
      href="/"
    >
      <img
        alt="Greg Rickaby"
        class="_logo_46c4bf"
        data-nimg="1"
        decoding="async"
        height="96"
        loading="eager"
        src="/_next/image?url=%2Fapp%2Ficon.png&w=256&q=75"
        srcset="/_next/image?url=%2Fapp%2Ficon.png&w=96&q=75 1x, /_next/image?url=%2Fapp%2Ficon.png&w=256&q=75 2x"
        style="color: transparent;"
        width="96"
      />
    </a>
    <div
      class="_brandingContainer_46c4bf"
    >
      <h1
        class="_siteTitle_46c4bf"
      >
        <a
          href="/"
        >
          Greg Rickaby
        </a>
      </h1>
      <p
        class="_siteDescription_46c4bf"
      >
        Cameras and Code
      </p>
    </div>
  </div>
  <nav
    class="_nav_2c0fa0"
    data-testid="nav"
  >
    <div
      class="_navContainer_2c0fa0"
    >
      <button
        aria-label="Toggle navigation"
        class="button _navButton_2c0fa0"
      >
        <svg
          class="tabler-icon tabler-icon-menu "
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 8l16 0"
          />
          <path
            d="M4 16l16 0"
          />
        </svg>
      </button>
    </div>
    <div
      class="_linkContainer_2c0fa0 hidden"
      data-testid="nav-links"
    >
      <a
        class="_link_2c0fa0"
        href="/about"
      >
        About
      </a>
      <a
        class="_link_2c0fa0"
        href="/blog"
      >
        Blog
      </a>
      <a
        class="_link_2c0fa0"
        href="/contact"
      >
        Contact
      </a>
      <a
        class="_link_2c0fa0"
        href="/cv"
      >
        CV
      </a>
      <a
        class="_link_2c0fa0"
        href="/media"
      >
        Media
      </a>
      <a
        class="_link_2c0fa0"
        href="https://www.threads.net/@gregoryrickaby"
      >
        Photos
      </a>
      <a
        class="_link_2c0fa0"
        href="/resources"
      >
        Resources
      </a>
      <a
        class="_link_2c0fa0"
        href="/feed.xml"
      >
        RSS
      </a>
    </div>
  </nav>
</header>
`;
